### DISCUSSION TITLE
Cursor functionality / theming

### DISCUSSION THREAD

**Discussion by cquinone** [MyreMylar/pygame_gui#262] [General]
Hi, 

I love this library, before I had made my own text boxes and GU elements per game, but this is general enough and well implemented so that I can use it in many games!

I use the text entry line element a lot, and I was wondering what can actually be controlled for the cursor - in the theming documentation I only see cursor color, but I'd specifically like to change things like cursor width and blink rate. Is that possible, even if not through a theme .JSON ?

**Comment by MyreMylar** [comment]
Hey there, 

Right now the timings for the text entry input are variables on the element. I hadn't anticipated anybody wanting to change the speed of the cursor blinking, but it should be an easy thing to add to theming in a future update. If you want to mess with the timing right now:

```
self.key_repeat = 0.5
self.cursor_blink_delay_after_moving = 1.0
self.blink_cursor_time = 0.4
```

Changing these three attributes of a created text entry line should let you play around with blinking and key entry speed.

Changing the width and height of the cursor is a bit more tricky at the minute, right now it is just hard coded to be 2 pixels wide on this line:

```
https://github.com/MyreMylar/pygame_gui/blob/main/pygame_gui/core/text/text_box_layout_row.py#L29
```

So you'd have to do something like:

```
my_entry_line.drawable_shape.text_box_layout.layout_rows[0].cursor_rect.width = 5
```

To change the size.

Not tested any of that yet so your mileage may vary.
